src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockPhysicsListener.java

```diff
@@ -112,4 +115,34 @@ public void onBucketUse(PlayerBucketEmptyEvent e) {
             e.setCancelled(true);
         }
     }
+
+    @EventHandler
+    public void onBlockUpdate(BlockPhysicsEvent event) {
+        if (!Slimefun.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_19)) {
+            // Method BlockData::isSupported is 1.19+
+            return;
+        }
+
+        Block block = event.getBlock();
+
+        // Listen for collapsing sensitive blocks
+        if (!block.getBlockData().isSupported(event.getBlock())) {
+            SlimefunItem sfItem = BlockStorage.check(block);
+
+            if (sfItem != null && !sfItem.useVanillaBlockBreaking()) {
+                // Drop items from surrounding sensitive blocks but don't drop from supporting block
+                event.setCancelled(true);
+                block.setType(Material.AIR, true);
+
+                for (ItemStack drop : sfItem.getDrops()) {
+                    if (drop != null && !drop.getType().isAir()) {
+                        block.getWorld().dropItemNaturally(block.getLocation(), drop);
+                    }
+                }
+
+                // Fixes #2944 - Don't forget to clear the Block Data
+                BlockStorage.clearBlockInfo(block);
+            }
+        }
+    }
```

Comment:
Not a huge fan of using this event at all due to the mass amounts of times its called and the lag that can be easily created by listening to it.

If we do go this route (which I'm not completely against just hesitant personally), we need to ensure that we do it as performantly as possible. Starting by re-arranging some of these filter ifs like doing the sf item checks before the supported check (as blockstorage checks are just checks to a map and are definitely more performant then the is supported method). 

We should also ensure we are only executing this code when the physics check calls for it, i.e. when the block is being broken because it's no longer supported, as there are definite cases we don't need to check at all.

Reply:
> Not a huge fan of using this event at all due to the mass amounts of times its called and the lag that can be easily created by listening to it.

Well I also thought like you but I ran a spark profiler on my local server and to my surprise it takes very little tick time.

> If we do go this route (which I'm not completely against just hesitant personally), we need to ensure that we do it as performantly as possible. Starting by re-arranging some of these filter ifs like doing the sf item checks before the supported check (as blockstorage checks are just checks to a map and are definitely more performant then the is supported method).

Done

> We should also ensure we are only executing this code when the physics check calls for it, i.e. when the block is being broken because it's no longer supported, as there are definite cases we don't need to check at all.

I don't understand what you mean by writing this.